iceberg/src/test/scala/org/apache/spark/sql/delta/serverSidePlanning/SparkToIcebergExpressionConverterSuite.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.serverSidePlanning
+
+import org.apache.spark.sql.sources._
+import org.scalatest.funsuite.AnyFunSuite
+import shadedForDelta.org.apache.iceberg.expressions.{Expression, ExpressionUtil, Expressions}
+
+class SparkToIcebergExpressionConverterSuite extends AnyFunSuite {
+
+  // Test schema used to bind expressions for semantic comparison.
+  // Binding resolves column names to field IDs/types, enabling type-safe comparison
+  // that catches bugs like "age < 30" (int) vs "age < "30"" (string)
+  private val testSchema = TestSchemas.defaultSchema.asStruct()
+
+  private def assertConvert(sparkFilter: Filter, expectedIcebergExpr: Expression): Unit = {
+    val result = SparkToIcebergExpressionConverter.convert(sparkFilter)
+    assert(result.isDefined, s"Should convert: $sparkFilter")
+
+    val actual = result.get
+    // ExpressionUtil.equivalent(left, right, struct, caseSensitive)
+    // caseSensitive=true: column names must match case exactly
+    assert(
+      ExpressionUtil.equivalent(expectedIcebergExpr, actual, testSchema, true),
+      s"Expressions not equivalent:\n  Expected: $expectedIcebergExpr\n  Actual: $actual"
+    )
+  }
+
+  private def assertReturnsNone(filter: Filter): Unit = {
+    val result = SparkToIcebergExpressionConverter.convert(filter)
+    assert(result.isEmpty, s"Should return None for unsupported filter: $filter")
+  }
+
+  // EqualTo tests
+  test("convert EqualTo with various types") {
+    assertConvert(EqualTo("id", 5), Expressions.equal("id", 5))  // integer
+    assertConvert(EqualTo("name", "Alice"), Expressions.equal("name", "Alice"))  // string
+    assertConvert(EqualTo("id", 1234567890L), Expressions.equal("id", 1234567890L))  // long
+    assertConvert(EqualTo("active", true), Expressions.equal("active", true))  // boolean
+
+    // EqualTo with null becomes IsNull
+    assertConvert(EqualTo("name", null), Expressions.isNull("name"))
+  }
+
+  // Comparison operator tests
+  test("convert comparison operators") {
+    assertConvert(LessThan("age", 30), Expressions.lessThan("age", 30))
```

Comment:
you are not testing with different types for these like you are doing for boolean?